Program.cs

```diff
@@ -1,16 +1,161 @@
+        }
+
+        #endregion
+
+        #region Task4
+
+        List<string> cities = new List<string>() { "Berlin", "London", "Paris", "Bobruisk", "Tokyo" };
+        var citiesfiltered = cities.Select(i => i).Where(i => i.StartsWith("B") && i.EndsWith("k")).ToList();
+
+        foreach (var g in citiesfiltered)
+        {
+            Console.WriteLine("City of dream is {0}", g);
+        }
+
+        #endregion
+
+        #region Task5
+        //couldn't solve, here my drafts
+
+        //Console.WriteLine("input string with words in different case");
+        //string allstring = Console.ReadLine();
+        //allstring.Substring(0,Indexof(Indexof()))
+        //   IsLower(allstring,)
+
+        //       if()
+        // List<string> allcases = new List<string>() { "BERLIN", "london", "paris", "BOBRUISK" ,"TOKYO" };
+
+        #endregion
+
+        #region Task6 
+        //couldn't solve yet, here drafts
+        int[] integers3 = new[] { 15, 4, -7, 15, -7, 14, 13, 4, 11, 15 };
+        /*var onlydistinct = integers3.Where(i => i.Distinct().ToList();
+        int[] reversedlist = onlydistinct.Reverse();
+        integers2.Distinct();*/
+
+        #endregion
+
+
+        #region Task7
+
+        bool checkIfTwo = IsTheNumberInMassive(2, integers);
+        Console.WriteLine($"Is  in massive?: {checkIfTwo}");
+
+        #endregion
+
+        #region Task8
+
+        /*List<string> twocolor = new List<string>();
+
+        List<string> tricolor = new List<string>() { "#FFFFFF", "FF0000", "#FFFFFF" };*/
+        //var selectedcolors = (tricolor.Count > 5).select(tricolor.First, tricolor.Last(a=> i)).ToList();
+
+        /*var selectedcolors = from t in tricolor 
+        where GetListAllElementsHeight(tricolor) > 5
+        select (tricolor.SelectMany(tricolor.First(), tricolor.Last()));*/// not sure how to select 1st and 3rd elements
+
+        //Console.WriteLine($"First and last colors are:{selectedcolors}");
+
+        #endregion
+
+        #region Task9
+
+        IList<Student> studentList = new List<Student>()
+        {
+            new Student() { StudentID = 1, StudentName = "John", Age = 18, StandardID = 1 },
+            new Student() { StudentID = 2, StudentName = "Steve", Age = 21, StandardID = 1 },
+            new Student() { StudentID = 3, StudentName = "Bill", Age = 18, StandardID = 2 },
+            new Student() { StudentID = 4, StudentName = "Ram", Age = 20, StandardID = 2 },
+            new Student() { StudentID = 5, StudentName = "Ron", Age = 21 }
+        };
+
+        var studentsNames = studentList.Select(s => s.StudentName).ToList();
+
+        #endregion
+    }
+
+    public static int GetListAllElementsHeight(List<string> list)
+    {
+        int listlength = 0;
+        for (int i = 0; i < list.Count - 1; i++)
+        {
+            listlength +=list[0].Length;
+        }
+        return listlength;
+    }
+    public static bool IsTheNumberInMassive(int numb, int[] integers)
```

Comment:
why do we need numb? it is not used in method

Reply:
fixed